content/docs/07-reference/ai-sdk-core/02-stream-text.mdx

```diff
@@ -590,6 +597,55 @@ for await (const textPart of textStream) {
             },
           ],
         },
+        {
+          type: 'TextStreamPart',
+          parameters: [
+            {
+              name: 'type',
+              type: "'tool-call-streaming-start'",
+              description:
+                'Indicates the start of a tool call streaming. Only available when streaming tool calls.',
+            },
+            {
+              name: 'toolCallId',
+              type: 'string',
+              description: 'The id of the tool call.',
+            },
+            {
+              name: 'toolName',
+              type: 'string',
+              description:
+                'The name of the tool, which typically would be the name of the function.',
+            },
+          ],
+        },
+        {
+          type: 'TextStreamPart',
+          parameters: [
+            {
+              name: 'type',
+              type: "'tool-call-delta'",
+              description:
+                'The type to identify the object as tool call delta. Only available when streaming tool calls.',
+            },
+            {
+              name: 'toolCallId',
+              type: 'string',
+              description: 'The id of the tool call.',
+            },
+            {
+              name: 'toolName',
+              type: 'string',
+              description:
+                'The name of the tool, which typically would be the name of the function.',
+            },
+            {
+              name: 'argsTextDelta',
+              type: 'string',
+              description: 'The text delta of the tool call arguments.',
+            },
+          ],
+        },
```

Comment:
do we want an end event, like `tool-call-stop` to know when the streaming ends?

Reply:
that's what `tool-call` already does (it comes when the streaming ends).